src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -356,6 +358,27 @@ private GetMappingsRequest buildGetMappingRequest(String indexName) {
         return getMappingsRequest;
     }
 
+    private static void validatePPLToolParameters(Map<String, Object> map) {
+        if (StringUtils.isBlank((String) map.get("model_id"))) {
+            throw new IllegalArgumentException("PPL tool needs non blank model id.");
+        }
+        if (map.containsKey("execute")) {
+            String execute = map.get("execute").toString().toLowerCase(Locale.ROOT);
```

Comment:
Is it possible the `map.get("execute")` is null? if yes, it could produce NPE here.